docs/source/reference/kubernetes/kubernetes-setup.rst

```diff
@@ -0,0 +1,223 @@
+
+
+.. _kubernetes-setup-eks:
+
+Deploying on AWS EKS
+^^^^^^^^^^^^^^^^^^^^
+
+1. Create a EKS cluster with at least 1 node. We recommend creating nodes with at least 4 vCPUs.
+
+2. TODO - Add instructions on getting kubeconfig.
+
+3. [If using GPUs] EKS clusters already come with Nvidia drivers setup. However, you will need to label the nodes with the GPU type. Use the SkyPilot node labelling tool to do so:
+
+    .. code-block:: console
+
+        python -m sky.utils.kubernetes.gpu_labeler
+
+
+   This will create a job on each node to read the GPU type from `nvidia-smi` and assign a ``skypilot.co/accelerator`` label to the node. You can check the status of these jobs by running:
+
+    .. code-block:: console
+
+        kubectl get jobs -n kube-system
+
+4. Verify your kubeconfig (and GPU support, if available) is correctly setup by running :code:`sky check`:
+
+    .. code-block:: console
+
+        $ sky check
+
+
+.. _kubernetes-setup-onprem:
+
+Deploying on On-Prem Clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also deploy Kubernetes on your On-Prem clusters using off-the-shelf tools,
+such as `kubeadm <https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/>`_,
+`k3s <https://docs.k3s.io/quick-start>`_ or
+`Rancher <https://ranchermanager.docs.rancher.com/v2.5/pages-for-subheaders/kubernetes-clusters-in-rancher-setup>`_.
+Please follow their respective guides to deploy your Kubernetes cluster.
+
+Setting up GPU support
+~~~~~~~~~~~~~~~~~~~~~~
+If your Kubernetes cluster has Nvidia GPUs, make sure you have the Nvidia
+device plugin installed (i.e., ``nvidia.com/gpu`` resource is available on each node).
+Additionally, you will need to label each node in your cluster with the GPU type.
+For example, a node with v100 GPUs must have a label :code:`skypilot.co/accelerators: v100`.
+
+We provide a convenience script that automatically detects GPU type and labels each node. You can run it with:
+
+.. code-block:: console
+
+ $ python -m sky.utils.kubernetes.gpu_labeler
+
+ Created GPU labeler job for node ip-192-168-54-76.us-west-2.compute.internal
+ Created GPU labeler job for node ip-192-168-93-215.us-west-2.compute.internal
+ GPU labeling started - this may take a few minutes to complete.
+ To check the status of GPU labeling jobs, run `kubectl get jobs --namespace=kube-system -l job=sky-gpu-labeler`
+ You can check if nodes have been labeled by running `kubectl describe nodes` and looking for labels of the format `skypilot.co/accelerators: <gpu_name>`.
+
+
+.. note::
+ GPU labelling is not required on GKE clusters - SkyPilot will automatically use GKE provided labels. However, you will still need to install `drivers <https://cloud.google.com/kubernetes-engine/docs/how-to/gpus#installing_drivers>`_.
+
+
+.. note::
+ To cleanup any leftover jobs from the GPU labelling process, run ``python -m sky.utils.kubernetes.gpu_labeler --cleanup``.
```

Comment:
At this point, wondering what are these leftover jobs.

Reply:
Thanks for catching - reworded now